Added weight vector to SwapOperator

kche309 opened the pull request:
I have added an optional integer weight vector to the SwapOperator. This will allow specifying a subset of the input parameter `RealParameter parameterInput` to be operated on. 

An example of xml usage:
```
<state id="state" spec="State" storeEvery="5000">
<parameter id="freqParameter" spec="parameter.RealParameter" dimension="4" lower="0.0" name="stateNode" upper="1.0">
0.25
</parameter>
</state>

...

<operator id="FrequenciesSwap" spec="SwapOperator" howMany="1" weight="0.01">
    <weightvector>0 1 0 1</weightvector>
    <parameter idref="freqParameter"/>
</operator>
```

rbouckaert commented:
@kche309 The name `weight` confused me a bit: the way the integer parameter is used looks like a `BooleanParameter` is more appropriate since it does not matter what value is used as long as it is not 0 the dimension will be included in the list of eligible dimensions. Maybe something like `filter` would be more appropriate?

kche309 commented:
@rbouckaert Yes, naming the parameter filter would be more appropriate. I have changed the type to `BooleanParameter` and renamed it to `filter`. The description of the `filter` now reads "filter to specify a subset of the parameter to operate on".
